modules/foreman_host_power.py

```diff
@@ -0,0 +1,167 @@
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_host_power
+short_description: Manage Foreman hosts power state
+description:
+  - "Manage power state of Foreman host"
+  - "This beta version can start and stop an existing foreman host and question the current power state."
+  - "Uses https://github.com/SatelliteQE/nailgun"
+version_added: "2.7"
+author:
+  - "Bernhard Hopfenmueller (@Fobhep) ATIX AG"
+requirements:
+  - "nailgun >= 0.29.0"
+  - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    required: false
+    default: true
+    type: bool
+  name:
+    description:
+      - name of host
+    required: true
+  domain_name:
+    description:
+      - name of host's domain
+    required: true
+  power_state:
+    description: Desired power state
+    type: list
+    choices:
+      - on
+      - off
+      - state
+'''
+
+EXAMPLES = '''
+- name: "Switch a host on"
+  foreman_host_power:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "test-host"
+    domain_name: "domain.test"
+    state: on
+
+- name: "Switch a host off"
+  foreman_host_power:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "test-host"
+    domain_name: "domain.test"
+    state: off
+
+- name: "Query host power state"
+  foreman_host_power:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "test-host"
+    domain_name: "domain.test"
+    state: state
+    register: result
+- debug:
+    msg: "Host power state is {{ result.power_state }}"
+
+
+'''
+
+RETURN = '''
+power_state:
+    description: current power state of host
+    returned: always
+    type: string
+    sample: "off"
+ '''
+
+try:
+    from nailgun.entities import (
+        Host,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        find_host,
+        naildown_power_state,
+        query_power_state,
+    )
+except ImportError:
+    pass
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'domain_name': 'domain_name',
+}
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            domain_name=dict(required=True),
+            state=dict(default='present', choices=['on', 'off', 'state']),
+        ),
+        supports_check_mode=True,
+    )
+
+    (host_dict, state) = module.parse_params()
+
+    module.connect()
+
+    host_dict['name'] = host_dict['name'] + '.' + host_dict['domain_name']
```

Comment:
I assume you've done this to be consistent with the other module? I personally think of hosts as `foo.example.com` as opposed to `foo` and therefore would prefer the `name` to take FQDN or just use a `hostname` parameter.

But, I'm not that bothered, so I'm happy either way if I'm honest.

Reply:
I guess, it's hard to support both?

Reply:
I agree with you @sean797, therefore I will change this parameter to hostname and document that one should use the FQDN.

Reply:
fixed